test/built-ins/Iterator/prototype/drop/return-is-forwarded.js

```diff
@@ -0,0 +1,51 @@
+// Copyright (C) 2023 Michael Ficarra. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-iteratorprototype.drop
+description: >
+  Underlying iterator return is called when result iterator is closed
+info: |
+  %Iterator.prototype%.drop ( limit )
+
+  6.c.iii. Let completion be Completion(Yield(? IteratorValue(next))).
+  6.c.iv. IfAbruptCloseIterator(completion, iterated).
+
+includes: [iterators.js]
+features: [iterator-helpers]
+flags: []
+---*/
+let returnCount = 0;
+
+class TestIterator extends Iterator {
+  next() {
+    return {
+      done: false,
+      value: 1
+    };
+  }
+  return() {
+    ++returnCount;
+    return {};
+  }
+}
+
+let iterator = new TestIterator().drop(0);
+assert.sameValue(returnCount, 0);
+iterator.return();
+assert.sameValue(returnCount, 1);
+iterator.return();
+assert.sameValue(returnCount, 1);
+
+iterator = new TestIterator().drop(1);
+assert.sameValue(returnCount, 1);
```

Comment:
Just as advice, I'd suggest resetting `returnCount` to zero in between these stanzas. I found out the hard way that I created a bunch of extra work for myself when I wrote tests like this for Temporal without resetting call counters, and then later had to add or remove stanzas when the semantics changed.